tools/code-generation/smithy/codegen/cpp-smoke-tests-codegen/src/main/java/com/amazonaws/util/awsclientsmithygenerator/generators/SmithyCodegenAdapter.java

```diff
@@ -0,0 +1,220 @@
+package com.amazonaws.util.awsclientsmithygenerator.generators;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.ShapeType;
+import software.amazon.smithy.model.shapes.MemberShape;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.Model;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class SmithyCodegenAdapter implements GenericCodegenAdapter<Shape, Node>{
```

Comment:
how does implementing `GenericCodegenAdapter` help us? i know before this in a previous iteration that it was between c2j and smithy, but it seems like a uneeded abstraction now where we could just use shape and node directly.

Reply:
So this really helps separate the implementation from the algorithm. It helps with abstraction